src/views/FarmAuction/hooks/useAuctionHistory.tsx

```diff
@@ -0,0 +1,44 @@
+import { useState, useEffect } from 'react'
+import { useFarmAuctionContract } from 'hooks/useContract'
+import { Auction, Bidder } from 'config/constants/types'
+import { AUCTION_BIDDERS_TO_FETCH } from 'config'
+import { processAuctionData, sortAuctionBidders } from '../helpers'
+
+interface AuctionHistoryMap {
+  [key: number]: {
+    auction: Auction
+    bidders: Bidder[]
+  }
+}
+
+const useAuctionHistory = (auctionId: number) => {
```

Comment:
This hook is used in the history panel. Similar to `useCurrentFarmAuction` but much simpler as it only needs to get historical data